src/main/java/org/osiam/auth/oauth_client/ClientDao.java

```diff
@@ -42,6 +42,11 @@ public ClientEntity getClient(final String id) {
         return getClientById(id);
     }
 
+    public List<ClientEntity> getClients() {
+        final Query query = em.createNamedQuery("getClients");
```

Comment:
Please use `javax.persistence.EntityManager#createNamedQuery(java.lang.String, java.lang.Class<T>)`. This way you get a `TypedQuery` and there will be no hassle with unchecked assignment in the line below.